packages/content-management/table_list_view_table/src/components/use_tag_filter_panel.tsx

```diff
@@ -45,15 +45,43 @@ export const useTagFilterPanel = ({
   addOrRemoveExcludeTagFilter,
   addOrRemoveIncludeTagFilter,
 }: Params) => {
+  const tags = useMemo(() => getTagList(), [getTagList]);
+
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
   // When the panel is "in use" it means that it is opened and the user is interacting with it.
   // When the user clicks on a tag to select it, the component is unmounted and mounted immediately, which
   // creates a new EUI transition "IN" which makes the UI "flicker". To avoid that we pass this
   // "isInUse" state which disable the transition.
   const [isInUse, setIsInUse] = useState(false);
   const [options, setOptions] = useState<TagOptionItem[]>([]);
-  const [tagSelection, setTagSelection] = useState<TagSelection>({});
-  const totalActiveFilters = Object.keys(tagSelection).length;
+  const clauseInclude: string[] = useMemo(
+    () =>
+      query ? toArray(query.ast.getOrFieldClause('tag', undefined, true, 'eq')?.value ?? []) : [],
+    [query]
+  );
+  const clauseExclude: string[] = useMemo(
+    () =>
+      query ? toArray(query.ast.getOrFieldClause('tag', undefined, false, 'eq')?.value ?? []) : [],
+    [query]
+  );
+  const [tagSelection, setTagSelection] = useState<TagSelection>({
+    ...clauseInclude.reduce((acc, tagName) => {
+      acc[tagName] = 'include';
+      return acc;
+    }, {} as TagSelection),
+    ...clauseExclude.reduce((acc, tagName) => {
+      acc[tagName] = 'exclude';
+      return acc;
+    }, {} as TagSelection),
+  });
+
+  const totalActiveFilters = useMemo(
+    () =>
+      tags.filter((currentOption) =>
+        [...clauseInclude, ...clauseExclude].find((clauseName) => clauseName === currentOption.name)
+      ).length,
+    [clauseExclude, clauseInclude, tags]
+  );
```

Comment:
Before:
<img width="2540" alt="incorrect_active_count" src="https://github.com/elastic/kibana/assets/6295984/2fe07160-1f9e-498e-a683-5d64729fc88b">


After:
https://github.com/elastic/kibana/assets/6295984/63fd6d2e-2a07-45c8-a624-a50ca8d8b3bc

Reply:
Reverted the change in this PR. Follow up in https://github.com/elastic/kibana/issues/160720